src/containers/BookList.js

```diff
@@ -18,29 +29,40 @@ const BookList = props => {
           <th>Category</th>
           <th>Remove book</th>
         </tr>
-        {books.map(book => (
-          <Book
-            bookObject={book}
-            key={UuidV4()}
-            handleClick={() => handleClick(book.id)}
-          />
-        ))}
+        {books
+          .filter(
+            book => filter === 'All' || filter === '' || book.category === filter,
+          )
```

Comment:
[Optional change] We can remove one conditional '**filter=== ' '** ' from the filter function. In the **CategoryFilter** component, you have added **All** optional with no value. You can pass **All** as a value in the option.